### PR TITLE
mistralai[patch]: 16k token batching logic embed

### PR DESCRIPTION
**Pull request by efriis**

### CODE REVIEW

File: libs/partners/mistralai/langchain_mistralai/embeddings.py [hunk 1]
```diff
@@ -72,8 +77,29 @@ def validate_environment(cls, values: Dict) -> Dict:
...
+            )
         return values
 
+    def _get_batches(self, texts: List[str]) -> Iterable[List[str]]:
+        """Split a list of texts into batches of less than 16k tokens
+        for Mistral API."""
+        batch: List[str] = []
+        batch_tokens = 0
+        for text in texts:
+            text_tokens = len(self.tokenizer.encode(text))
```
Comment (hinthornw) [inline libs/partners/mistralai/langchain_mistralai/embeddings.py:92]: Should we `batch_encode()`?

File: libs/partners/mistralai/langchain_mistralai/embeddings.py [hunk 2]
```diff
@@ -84,13 +110,17 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
             List of embeddings, one for each text.
         """
         try:
-            embeddings_batch_response = self.client.embeddings(
-                model=self.model,
-                input=texts,
+            batch_responses = (
+                self.client.embeddings(
```
Comment (hinthornw) [inline libs/partners/mistralai/langchain_mistralai/embeddings.py:114]: Do we have no mapping from source doc to embedding at this point if we're flat mapping the outputs? that seems undesirable. I'd probably average the embeddings or if we had been able to change the return type, return objects

File: libs/partners/mistralai/langchain_mistralai/embeddings.py (hunk 1 above)
Comment (efriis) [reply libs/partners/mistralai/langchain_mistralai/embeddings.py]: done!

File: libs/partners/mistralai/tests/integration_tests/test_embeddings.py [hunk 3]
```diff
@@ -38,10 +39,10 @@ async def test_mistralai_embedding_query_async() -> None:
 
 def test_mistralai_embedding_documents_long() -> None:
     """Test MistralAI embeddings for documents."""
-    documents = ["foo bar " * 1000, "test document " * 1000]
+    documents = ["foo bar " * 1000, "test document " * 1000] * 5
```
Comment (efriis) [inline libs/partners/mistralai/tests/integration_tests/test_embeddings.py:42]: this test would fail before

**Review by hinthornw — APPROVED** [review]

**Pull request merged**